src/core/message-parser.ts

```diff
@@ -10,48 +10,119 @@ interface MessagePayload {
   clientId?: string;
   timestamp: number;
   extras?: {
-    timeserial?: string;
     headers?: MessageHeaders;
   };
+
+  serial: string;
+  updatedAt?: number;
+  deletedAt?: number;
+  action: Ably.MessageAction;
+  operation?: Ably.Operation;
+}
+
+interface ChatMessageFields {
+  timeserial: string;
+  clientId: string;
+  roomId: string;
+  text: string;
+  createdAt: Date;
+  metadata: MessageMetadata;
+  headers: MessageHeaders;
+  deletedAt?: Date;
+  deletedBy?: string;
+  deletionDetail?: MessageDetails;
+  updatedAt?: Date;
+  updatedBy?: string;
+  updateDetail?: MessageDetails;
 }
 
-export function parseMessage(roomId: string | undefined, message: Ably.InboundMessage): Message {
-  const messageCreatedMessage = message as MessagePayload;
+export function parseMessage(roomId: string | undefined, inboundMessage: Ably.InboundMessage): Message {
+  const message = inboundMessage as MessagePayload;
   if (!roomId) {
     throw new Ably.ErrorInfo(`received incoming message without roomId`, 50000, 500);
   }
 
-  if (!messageCreatedMessage.data) {
+  if (!message.data) {
     throw new Ably.ErrorInfo(`received incoming message without data`, 50000, 500);
   }
 
-  if (!messageCreatedMessage.clientId) {
+  if (!message.clientId) {
     throw new Ably.ErrorInfo(`received incoming message without clientId`, 50000, 500);
   }
 
-  if (!messageCreatedMessage.timestamp) {
+  if (!message.timestamp) {
     throw new Ably.ErrorInfo(`received incoming message without timestamp`, 50000, 500);
   }
 
-  if (messageCreatedMessage.data.text === undefined) {
+  if (message.data.text === undefined) {
     throw new Ably.ErrorInfo(`received incoming message without text`, 50000, 500);
   }
 
-  if (!messageCreatedMessage.extras) {
+  if (!message.extras) {
     throw new Ably.ErrorInfo(`received incoming message without extras`, 50000, 500);
   }
 
-  if (!messageCreatedMessage.extras.timeserial) {
+  if (!message.serial) {
     throw new Ably.ErrorInfo(`received incoming message without timeserial`, 50000, 500);
   }
 
-  return new DefaultMessage(
-    messageCreatedMessage.extras.timeserial,
-    messageCreatedMessage.clientId,
+  let operationDetails: MessageDetails | undefined;
```

Comment:
Do we need any validation around the operation?

Reply:
I don't think so, these are coming from realtime, so should be handle by ably-js. As to if they are present, both fields are optional, so we don't need to check that either.

Reply:
Sure ok - though I think it should only `trace` or `debug` log. Warnings are things that indicate an issue, and us receiving an event we don't know how to handle, is more likely to be non-issue imo

Reply:
Okay, I've made it so the filter will only emit a debug log if the event is not something we handle yet.